lsp/client/visualStudio/IdesLspPoc/LspClient/CloudFormationLspClient.cs

```diff
@@ -0,0 +1,39 @@
+﻿using IdesLspPoc.ContentDefinitions;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Utilities;
+using System.ComponentModel.Composition;
+
+namespace IdesLspPoc.LspClient
+{
+    // NOTE : We use the SDK and will need to split code/references across major VS versions
+
+    // Design thoughts - one of these for each distinct LSP we manage. We could have multiple ContentType declarations.
+
+    // this one "just works" to associate this LSP with yaml files
+    [ContentType("yaml")]
+
+    // for some reason, we have associate LSPs with JSON using a custom type - see JsonContentType for comments
+    [ContentType(JsonContentType.ContentTypeName)]
+
+    // [ContentType(BuildSpec.ContentType)] // Only add this if you're supporting custom file extensions
+    [Export(typeof(ILanguageClient))]
+    public class CloudFormationLspClient : ToolkitLspClient
```

Comment:
Given the ContentType declarations here match that of the BuildspecLspClient above, how does VS know which to pick or does the VS extension code know and explicitly instantiate the correct one?  I'm guessing with the use of ExportAttribute, this may not be the latter case.

Reply:
Both classes are registering their respective language server to be associated with json and yaml files. Visual Studio handles multiple languages servers getting concurrently registered to a file type.